src/components/day/Day.js

```diff
@@ -295,13 +295,43 @@ export default class DayComponent extends Field {
       this.addEventListener(this.refs.input, this.info.changeEvent, () => this.updateValue(null, {
         modified: true
       }));
+      this.addFocusBlurEvents();
     }
+    this.restoreFocus();
     this.setValue(this.dataValue);
     // Force the disabled state with getters and setters.
     this.disabled = this.shouldDisabled;
     return superAttach;
   }
 
+  addFocusBlurEvents() {
+    this.addEventListener(this.refs.day, 'focus', () => {
+      this.focusedElement = this.refs.day;
+    });
+    this.addEventListener(this.refs.day, 'blur', () => {
+      this.focusedElement = '';
+    });
+    this.addEventListener(this.refs.year, 'focus', () => {
+      this.focusedElement = this.refs.year;
+    });
+    this.addEventListener(this.refs.year, 'blur', () => {
+      this.focusedElement = '';
+    });
+    this.addEventListener(this.refs.month, 'focus', () => {
+      this.focusedElement = this.refs.month;
+    });
+    this.addEventListener(this.refs.month, 'blur', () => {
+      this.focusedElement = '';
+    });
```

Comment:
The same code repeats 3 times. Please create a method that will accept the ref and attach focus and blur events for it.